Composer/packages/adaptive-form/src/components/fields/BooleanField.tsx

```diff
@@ -28,9 +28,38 @@ const BooleanField: React.FC<FieldProps> = function CheckboxWidget(props) {
     },
   ];
 
+  if (expression) {
+    options.push(
+      {
+        key: 'divider',
+        text: '-',
+        itemType: DropdownMenuItemType.Divider,
+      },
+      {
+        key: 'expression',
+        text: formatMessage('Write in expression'),
+      }
+    );
+  }
+
   const handleChange = (e, option?: IDropdownOption) => {
     if (option) {
-      const optionValue = option.key === 'none' ? undefined : option.key === 'true';
+      let optionValue: boolean | string | undefined;
+      switch (option.key) {
+        case 'none':
+          optionValue = undefined;
+          break;
+        case 'true':
+          optionValue = true;
+          break;
+        case 'false':
+          optionValue = false;
+          break;
+        case 'expression':
+          optionValue = '=';
+          break;
+      }
```

Comment:
Because optionValue is multiple possible types, I would have a default: case that either throws and error or assigns a default value.  It isn't super clear what optionValue ends up being if no case matches: false or undefined.

Reply:
The options are listed above. The switch only go over those, there shouldn't be _no case_ match. If someone inevertently change the code to have a mismatch between options and handleChange, optionValue would be undefined which is a valid value to pass up to the parent handler.

![image](https://user-images.githubusercontent.com/66701106/96322319-6f014a80-0fcd-11eb-8036-0154b3fa1d6f.png)